### ISSUE TITLE
Frequently I get this error when I run test using protractor

### ISSUE THREAD

**Issue by mehdhi** [jasmine/jasmine#1491]
This issue occurs frequently but randomly, sometimes the test passes. 

```webpack: Compiled successfully.
[16:05:16] I/update - chromedriver: file exists \c\someproject\node_modules\protractor\node_modules\webdriver-manager\selenium\chromedriver_2.35.zip
[16:05:16] I/update - chromedriver: unzipping chromedriver_2.35.zip
[16:05:16] I/update - chromedriver: chromedriver_2.35.exe up to date
[16:05:16] I/launcher - Running 1 instances of WebDriver
[16:05:16] I/direct - Using ChromeDriver directly...

DevTools listening on ws://127.0.0.1:12270/devtools/browser/39dae962-b401-4bd0-a9e0-5d79c4886a3a
Jasmine started
Mock Server started
\c\someproject\node_modules\jasmine\node_modules\jasmine-core\lib\jasmine-core\jasmine.js:5277
    return children && children[0].result.status;
                                  ^

TypeError: Cannot read property 'result' of undefined
    at isAfterAll (\c\someproject\node_modules\jasmine\node_modules\jasmine-core\lib\jasmine-core\jasmine.js:5277:35)
    at Suite.onException (\c\someproject\node_modules\jasmine\node_modules\jasmine-core\lib\jasmine-core\jasmine.js:5240:8)
    at Suite.onException (\c\someproject\node_modules\jasmine\node_modules\jasmine-core\lib\jasmine-core\jasmine.js:5252:27)
    at Suite.onException (\c\someproject\node_modules\jasmine\node_modules\jasmine-core\lib\jasmine-core\jasmine.js:5252:27)
    at QueueRunner.onException (\c\someproject\node_modules\jasmine\node_modules\jasmine-core\lib\jasmine-core\jasmine.js:5490:34)
    at onException (\c\someproject\node_modules\jasmine\node_modules\jasmine-core\lib\jasmine-core\jasmine.js:4461:14)
    at handleError (\c\someproject\node_modules\jasmine\node_modules\jasmine-core\lib\jasmine-core\jasmine.js:4394:11)
    at process.onerror (\c\someproject\node_modules\jasmine\node_modules\jasmine-core\lib\jasmine-core\jasmine.js:2455:17)
    at emitOne (events.js:115:13)
    at process.emit (events.js:210:7)
    at process.emit (\c\someproject\node_modules\source-map-support\source-map-support.js:439:21)
    at process._fatalException (bootstrap_node.js:329:26)
npm ERR! code ELIFECYCLE
npm ERR! errno 7
npm ERR! someproject@0.0.0 e2e: `node ./.npm-scripts/serve.js && ng e2e --port 4200`
npm ERR! Exit status 7
npm ERR!
npm ERR! Failed at the someproject@0.0.0 e2e script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     \c\Users\FOOBAR\AppData\Roaming\npm-cache\_logs\2018-02-01T10_35_23_646Z-debug.log
```

## Error is thrown due to a missing null check for the first element of children array
```
function isAfterAll(children) {
    return children && children[0].result.status;
 }
```
Path: node_modules\jasmine\node_modules\jasmine-core\lib\jasmine-core\jasmine.js
Line: 5277

## Your Environment

Version 64.0.3282.119 (Official Build) (64-bit)
Windows 10 Pro
npm@5.1.0
node@v8.1.4
jasmine-core@2.9.1

**Comment by slackersoft** [comment]
This error is most likely a leak of a previous async spec that has told Jasmine to move on, but still has code executing in the background. It looks like some asynchronous code is throwing an Error that Jasmine is trying to associate with the suite/spec that is currently running and it's finding a suite that doesn't have any children. Since Jasmine doesn't actually execute any user code in suites without children (reporter stuff still happens), the assumption is that if `onException` gets invoked on a suite that it must have children.

This code should be fixed on master now as part of Jasmine 3.0. Jasmine no longer tries to determine whether an `afterAll` is running anymore and just logs all errors that occur in `beforeAll` or `afterAll` directly on the suite itself.

Hope this helps. Thanks for using Jasmine!

**Comment by mehdhi** [comment]
@slackersoft Thanks alot.
